### ISSUE TITLE
wrong warning on ambiguous image files

### ISSUE THREAD

**Issue by svigerske** [doxygen/doxygen#7189]
If the IMAGE_PATH contains a directory twice, one may get a warning about the image file name being ambiguous, even though the presented candidates are actually all the same file.

The example [imageambiguous.zip](https://github.com/doxygen/doxygen/files/3496401/imageambiguous.zip) illustrates the issue with doxygen 1.8.16 (was fine with 1.8.15).
It has `IMAGE_PATH = img ./img` and this generates (twice) the warning
```
/some/long/path/main.md:2: warning: image file name doxygen.png is ambiguous.
Possible candidates:
   /some/long/path/img/doxygen.png
   /some/long/path/img/doxygen.png
```

Cleaning up the IMAGE_PATH works around this, of course, but to me it still looks as if the warning is wrong.

**Comment by albert-github** [comment]
**Some background**
The fact that doxygen warns for ambiguous images is good as when different images with the same name exist the wrong image could be used and the user would not notice it except when going through the result output. Also when the same image is on different directories this is a potential problem.

**Some analysis**
When looking at other PATH specific inputs, we see that with `INPUT` when specifying a directory twice a file is only added once. With the `IMAGE_PATH` the file is added twice.
Looking in the code this looks like to be due to the fact that in case of `INPUT` the call to `readFileOrDirectory` has the `killDict` argument and in case of `IMAGE_PATH` this is missing.
A small test shows that when adding `killDict` for `IMAGE_PATH` the message is gone.

@doxygen 
- are there any side effects in adding `killDict` for `IMAGE_PATH` expected?
- should the `killDict` be added for all `readFileOrDirectory` calls?

**Comment by svigerske** [comment]
I agree that the warnings are good. For us it pointed to some issue where we had files with the same name in different directories.

The actual cause of the problem reported in this issue was from having one image path being a subdirectory of another one and having RECURSIVE enabled, e.g.,
```
RECURSIVE = YES
IMAGE_PATH = somedir somedir/img
```
I don't know whether the mentioned `killDict` argument would also handle this.

**Comment by doxygen** [comment]
Indeed, `killDict` can be used to filter out duplicate directories and should indeed be used for all user provided directories. I've pushed a commit for this.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
